packages/next/src/build/webpack/plugins/flight-client-entry-plugin.ts

```diff
@@ -1039,23 +1049,33 @@ function addClientImport(
     isCjsModule ? 'commonjs' : 'auto'
   )
 
-  const isAutoModuleSourceType = assumedSourceType === 'auto'
-  if (isAutoModuleSourceType) {
-    clientComponentImports[modRequest] = new Set(['*'])
+  const clientImportsSet = clientComponentImports[modRequest]
+
+  if (importedIdentifiers[0] === '*') {
+    // If there's collected import path with named import identifiers,
```

Comment:
I don’t follow. If there’s **no** named import, but only an `import *`, shouldn’t we do the same?

Reply:
yea, in the outgoing connections. I amended a comment here. `[...clientImportsSet][0] !== '*'` covers both two case, it can either represent empty client imports or containing named imports.

Also pushed a test, added more explanation in the PR description last section